[DOC] Add docs and CHANGELOG record for disabling ownerReference on CA secrets (#3933)
strimzi/strimzi-kafka-operator: documentation/modules/security/ref-certificates-and-secrets.adoc

It needs to be accessible to the Strimzi components and for administrative access if you are intending to issue application certificates without using the User Operator.
You can enforce this using Kubernetes role-based access controls if necessary.

== Disabling `ownerReference` in the CA Secrets

> By default, the CA secrets created by Strimzi will have the `ownerReference` field set to the `Kafka` custom resource.

Reviewer: ```suggestion
By default, the CA secrets created by Strimzi have the `ownerReference` field set to the `Kafka` custom resource.
```

> This means that when the custom resource is deleted, the CA secrets will be garbage collected (deleted) by Kubernetes as well.

Reviewer: ```suggestion
This means that when the `Kafka` custom resource is deleted, the CA secrets are garbage collected (deleted) by Kubernetes as well.
```

> If you want to re-use the CA for a new cluster, you can disable the `ownerReference`.

Reviewer: ```suggestion
If you want to reuse the CA for a new cluster, you can disable the `ownerReference`.
```

> When disabled, the CA secrets will not be deleted by Kubernetes together with the corresponding `Kafka` custom resource.

Reviewer: ```suggestion
When disabled, CA secrets are not deleted by Kubernetes when the corresponding `Kafka` custom resource is deleted.
```

> .Example Kafka configuration with disabled `ownerReference` for Cluster and Client CAs

Reviewer: ```suggestion
.Example Kafka configuration with disabled `ownerReference` for cluster and client CAs
```
